### PR TITLE
[FIX] account: do not call action_invoice_paid

### PR DESCRIPTION
**Pull request by nim-odoo**
`action_invoice_paid` should be called when the computed field
`invoice_payment_state` is set. However, the recomputation doesn't go
through `create` or `write` (since the field is protected).

A workaround  is to call it at reconciliation. Indeed, an invoice can
only be paid when it is reconciled. For invoices with an amount of zero,
we call the method at posting.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qdp-odoo** [comment]
Diff looks okay, yet I didn't test it :-s

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2138,6 +2127,12 @@ def post(self):
             else:
                 continue
 
+        # Trigger action for paid invoices in amount is zero
+        self.filtered(
+            lambda m: m.is_invoice(include_receipts=True)
+            and not m.amount_total
+        ).action_invoice_paid()
```
Comment (smetl) [inline addons/account/models/account_move.py]: `and m.currency_id.is_zero(amount_total)`
plz inline

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -3651,6 +3652,12 @@ def reconcile(self, writeoff_acc_id=False, writeoff_journal_id=False):
             remaining_moves = (remaining_moves + writeoff_to_reconcile).auto_reconcile_lines()
         # Check if reconciliation is total or needs an exchange rate entry to be created
         (self + writeoff_to_reconcile).check_full_reconcile()
+
+        # Trigger action for paid invoices
+        not_paid_invoices.filtered(
+            lambda m: m.invoice_payment_state in ('paid', 'in_payment')
+        ).action_invoice_paid()
```
Comment (smetl) [inline addons/account/models/account_move.py]: plz inline

File: addons/account/models/account_move.py [hunk 3]
```diff
@@ -3632,6 +3627,12 @@ def reconcile(self, writeoff_acc_id=False, writeoff_journal_id=False):
         if not self:
             return
 
+        # List unpaid invoices
+        not_paid_invoices = self.filtered(
+            lambda ml: ml.move_id.is_invoice(include_receipts=True)
+            and ml.move_id.invoice_payment_state not in ('paid', 'in_payment')
+        ).mapped('move_id')
```
Comment (smetl) [inline addons/account/models/account_move.py]: More efficient to map before filtering

**Comment by nim-odoo** [comment]
robodoo r+

**Comment by smetl** [comment]
@robodoo r-
@nim-odoo Don't skip my review dude...

**Comment by nim-odoo** [comment]
@smetl it was taken into account, except the purely cosmetic changes (it's cosmetic, therefore subjective).

robodoo r+

**Comment by nim-odoo** [comment]
my bad, I missed on of them.

robodoo r+

**Pull request closed without merging**